### DISCUSSION TITLE
Restrict access to Kafka topics/tables

### DISCUSSION THREAD

**Discussion by tomkrol** [trinodb/trino#18969] [Ideas]
Currently when the Kafka catalog is configured with property :

kafka.table-description-supplier=CONFLUENT

then all topics from Schema Registry will be listed when you run ;  

SHOW TABLES IN catalog.schema

There is no way to limit scope of the tables. Would be good to have such a property (e.g. kafka.table-names), that will limit access only to specific tables. 

Restricting access using certs doesn't fit to all use cases.

**Comment by hashhar** [comment]
See https://trino.io/docs/current/security/built-in-system-access-control.html, you can either implement your own or use https://trino.io/docs/current/security/file-system-access-control.html to limit what is visible.

**Reply by tomkrol** [reply]
Ah you;re right, didnt notice you can control table access via rules, too. 

Cheers
